src/source.rb

```diff
@@ -0,0 +1,17 @@
+require_relative './item'
+
+class Source < Item
+  attr_accessor :name, :items
+  attr_reader :id
+
+  def initialize(name)
+    @id = Random.rand(1..1000)
+    @name = name
+    @items = []
+    super(@id, nil, nil, nil, nil)
+  end
+
+  def add_item(item)
+    @items << item
+  end
```

Comment:
- [ ] good job here, just the @item.source should be set to self.

Reply:
Thank you @mikethreels , we will add this as well.